jersey-servers/src/main/java/com/palantir/remoting2/servers/jersey/RemoteExceptionMapper.java

```diff
@@ -30,29 +29,35 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+/**
+ * This {@link ExceptionMapper} is used when forwarding an exception from a remote server back to a calling client.
+ *
+ * In the following call stack, where a browser calls a server A which calls another server B:
+ *
+ * caller/browser -> [server A] -> [server B]
+ *
+ * this is the exception mapper used in Server A. When code in B throws an exception, the {@link JsonExceptionMapper}
+ * maps that exception to a JSON response with appropriate HTTP status code and returns it to server A over HTTP.
+ * That response is then processed by a {@link com.palantir.remoting2.errors.SerializableErrorToExceptionConverter} and
+ * thrown as a {@link RemoteException} at the call point in server A.  If server A does not catch this exception and
+ * it raises up the call stack back into Jersey, execution enters this {@link RemoteExceptionMapper}.
+ *
+ * To preserve debuggability, the exception and HTTP status code from B's exception are logged at WARN level and
+ * returned back to the caller/browser.
+ */
 @Provider
 final class RemoteExceptionMapper implements ExceptionMapper<RemoteException> {
 
     private static final Logger log = LoggerFactory.getLogger(RemoteExceptionMapper.class);
 
     @Override
     public Response toResponse(RemoteException exception) {
-        String errorId = UUID.randomUUID().toString();
-
         Status status = Status.fromStatusCode(exception.getStatus());
 
-        // Log responses that indicate client error (4xx) at higher level than server error (5xx)
-        if (Status.Family.CLIENT_ERROR.equals(status.getFamily())) {
-            log.error("Received response status code {} from server handling request. errorId: {}",
-                    SafeArg.of("statusCode", status.getStatusCode()),
-                    SafeArg.of("errorId", errorId),
-                    exception);
-        } else {
-            log.info("Received response status code {} from server handling request. errorId: {}",
-                    SafeArg.of("statusCode", status.getStatusCode()),
-                    SafeArg.of("errorId", errorId),
-                    exception);
-        }
+        // log at WARN instead of ERROR because although this indicates an issue in a remote server, it is not
+        log.warn("Forwarding response and status code {} from remote server back to caller",
```

Comment:
added.

it's propagated via the exception message, but not in a structured way right now